content/en/config.md

```diff
@@ -13,7 +13,7 @@ A litmus test for whether an app has all config correctly factored out of the co
 
 Note that this definition of "config" does **not** include internal application config, such as `config/routes.rb` in Rails, or how [code modules are connected](http://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html) in [Spring](http://spring.io/).  This type of config does not vary between deploys, and so is best done in the code.
 
-Another approach to config is the use of config files which are not checked into revision control, such as `config/database.yml` in Rails.  This is a huge improvement over using constants which are checked into the code repo, but still has weaknesses: it's easy to mistakenly check in a config file to the repo; there is a tendency for config files to be scattered about in different places and different formats, making it hard to see and manage all the config in one place.  Further, these formats tend to be language- or framework-specific.
+Another approach to config is the use of config files which are not checked into revision control, such as `config/database.yml` in Rails.  This is a huge improvement over using constants which are checked into the code repo.  And it can be more secure, such as in an environment where environment variables might get logged and logs might get sent to other services.  But it also has weaknesses: it's easy to mistakenly check in a config file to the repo; there is a tendency for config files to be scattered about in different places and different formats, making it hard to see and manage all the config in one place.  Further, these formats tend to be language- or framework-specific.
```

Comment:
What's stopping you from logging the configuration file contents too?

Reply:
Not bulletproof, but equivalent to the safety argument going the other way...

The arguments are that, in general or from a less knowledgeable developer, you're much less likely to end up logging the contents of the file, whereas you might inadvertently set up a system that logs out the environment contents.

The argument of "it's safer because you're less likely to expose the creds via logging" deserves to be on the same footing with "it's safer because you're less likely to check the creds into version control"... both are decent arguments in practical matters; neither is a for-sure thing.